W-14915457 rpa setup guide

tasmoore opened the pull request.

tasmoore committed:
Create rpa-setup-guide.adoc

tasmoore committed:
Update nav.adoc

tasmoore committed:
Update nav.adoc

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,145 @@
++
+image placeholder 1
++
+image placeholder 2
```
Hi @tasmoore, you only need one of each duplicated image. If you like them, you can use the marked-up screenshots. If not, use the blank ones and add your own mark-up.

tasmoore committed:
image updates

tasmoore committed:
added activity 1 and 2 images

tasmoore committed:
added activity 3 content and images

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
update numbering and add images

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
Update rpa-setup-guide.adoc

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+
+== Before You Begin
+
+* Create an Anypoint Platform account xref:rpa-home::automation-userrolesandpermissions-anypointrpapermissions.adoc[and assign it the `RPA Automations Manager` and `RPA Bots Manager` permissions].
```
Can you just create an Anypoint Platform Account? Don't you have to buy the Automation bundle to use this solution?

tasmoore replied:
Good question! Are we assuming the customer will have access to create an Anypoint account?

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+== Before You Begin
+
+* Create an Anypoint Platform account xref:rpa-home::automation-userrolesandpermissions-anypointrpapermissions.adoc[and assign it the `RPA Automations Manager` and `RPA Bots Manager` permissions].
+* Download the latest version of MuleSoft RPA Manager, RPA Builder, and RPA Bot.
```
RPA Manager is a cloud solution that belongs to the MuleSoft Automation bundle, so this is provided and has not to be installed. You have to download RPA Builder and RPA Bot from the RPA Manager Home Page and then connect them to your RPA Manager instance. 

- https://docs.mulesoft.com/rpa-builder/latest/install-and-configure
- https://docs.mulesoft.com/rpa-bot/latest/installation

sebermuley replied:
You have to installl Bot and Builder on the computer from where you can access the AMS.

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+* Download the latest version of MuleSoft RPA Manager, RPA Builder, and RPA Bot.
+* Ensure you have access to your AMS and its database metadata.
+* Ensure you have access to the Salesforce Financial Services Cloud organization to which your AMS data will be loaded.
+* Connect the RPA Bot that is built for this solution to RPA Manager.
```
Remove this sentence. The information should be included in line 10. (Bots are not built.)

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+== Adapting the CRPA File to Your AMS
+
+. Double-click each activity on the model pane to open the workflow.
+. Compare the shown process with your process and adapt the action steps accordingly. See How to adapt the CRPA file for an example.
```
Missing link? How to adapt...

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+
+== Transferring Project to the Test Phase and Testing It
+
+Now that you have adapted the file to your AMS, xref:rpa-manager::processautomation-deploy.adoc#test-a-process-automation[follow these steps] to transfer the project to the Test phase and test it.
```
Replace the link text 'follow these steps' with 'transfer the project to the Test phase and test it'.

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+
+== Transferring the Project to the Production Phase
+
+After successfully testing the project, xref:rpa-manager::processautomation-deploy.adoc[transfer the project to the Production phase]. Once transferred, do one of the following:
```
Remove link, because the phase transition is part of each of the two options.
(We still need to clarify whether to explain both options or choose one. )

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+After successfully testing the project, xref:rpa-manager::processautomation-deploy.adoc[transfer the project to the Production phase]. Once transferred, do one of the following:
+
+* xref:rpa-manager::processautomation-deploy.adoc#deploy-a-process-automation[Deploy the project] and schedule it as needed.
+* Publish the project to Anypoint Exchange as an invocable run configuration, which can be called as an invocable action in a flow.
```
Use link from line 51 with 'Publish the project to Anypoint Exchange as an invocable run configuration' as link text.

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+
+* xref:rpa-manager::processautomation-deploy.adoc#deploy-a-process-automation[Deploy the project] and schedule it as needed.
+* Publish the project to Anypoint Exchange as an invocable run configuration, which can be called as an invocable action in a flow.
+** xref:rpa-manager::processautomation-deploy#publish-automation-exchange[Publish RPA Manager Project to Exchange]
```
Remove because included in text above.

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+** xref:rpa-manager::processautomation-deploy#publish-automation-exchange[Publish RPA Manager Project to Exchange]
+** xref:rpa-manager::settings-connect.adoc[RPA Manager Settings]
+** https://help.salesforce.com/s/articleView?id=sf.flow_build_use_flows_with_mulesoft_rpa.htm&type=5[Use Flows with MuleSoft RPA]
+** xref:composer::ms_composer_rpa_reference.adoc[RPA Reference]
```
We need to check if we are supposed to use Composer at all for this solution. If not, remove link.

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+
+=== Activity 1: Log in to the AMS
+
+Follow these steps to log in to the AMS:
```
These is actually an overview of the  steps contained in the .crpa file. 
Generally there are 4 steps to log in to the AMS. These are marked with the numbered boxes 1 to 4, which can contain more than one Action Step to implement the one general step.

So instead of writing 'Follow these steps', maybe use an additional sub-heading like 'Overview of the Activity Workflow' and another sub-heading of the same level here: ***

tasmoore replied:
I am not sure if I applied your change correctly, so please take a look at the revision.

sebermuley replied:
Hi Tasha, 
I thinks that's clearer. Could you adpapt the second (and probably third) Activity using these headings, too?

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+. Select the database using another in-going Activity Parameter.
++
+image::rpa-setup-activity-parameters-wizard.png[RPA Builder Activity Parameter Wizard]
+
```
*** Add another heading like 'Minimum steps to change/adapt in this Workflow' here and restart the numbering.

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+. Change the following in the *Run Program Wizard* dialog:
+
+.. Replace the values in the *File name* and *Window Title* fields with the values from your AMS.
+.. Click the *Test* button to run the application so that you can use it to adapt the following Action Steps:
```
Replace : with .

tasmoore replied:
I think the : is needed since "the following" is used in the sentence.

sebermuley replied:
Sorry, I think I misplaced  this comment...

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
++
+image::rpa-setup-activity-parameters-wizard.png[RPA Builder Activity Parameter Wizard]
+
+. Change the following in the *Run Program Wizard* dialog:
```
Replace 
'*Run Program Wizard* dialog'
with
'xref:https://docs.mulesoft.com/rpa-builder/latest/toolbox-control-run-program.adoc[*Run Program* Action Step]'

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+
+. Change the following in the *Run Program Wizard* dialog:
+
+.. Replace the values in the *File name* and *Window Title* fields with the values from your AMS.
```
the *File name*, *Parameters*, and *Window Title* fields

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
++
+image::rpa-setup-run-program-wizard.png[RPA Builder Run Program Wizard]
+
+. Use the *Start from scratch* option to capture and test a new pattern that indicates the login page of the AMS is visible onscreen:
```
Use the *Start from scratch* option in the https://docs.mulesoft.com/rpa-builder/latest/toolbox-checks-image-search[*Image Search* Action Step]

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
++
+image::rpa-setup-image-search-wizard.png[RPA Builder Image Search Wizard]
+
+. Re-identify the field to enter the password. It may be necessary to add a similar Action Step above for entering the user name.
```
in the https://docs.mulesoft.com/rpa-builder/latest/toolbox-app-automation-keystrokes-to-app-element[*Keystrokes to App Element* Action Step]

sebermuley replied:
(There might be more than one Action Step of the same type. But the one referred to here is marked in the screenshot. Optionally we could also add the individual name of the AS of the example.)

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
++
+image::rpa-setup-keystrokes-app-element-wizard.png[RPA Builder Keystrokes to App Element Wizard]
+
+. Re-identify the *Login* button to click on:
```
in the https://docs.mulesoft.com/rpa-builder/latest/toolbox-app-automation-click-app-element[*Click App Element* Action Step]

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+Change the following Action Steps:
+
+. Adapt the first Action Steps as you did in the previous Activity.
+. Adapt the scan area in the AI OCR Action Step in which RPA Bot searches for the record entry with the lookup code:
```
https://docs.mulesoft.com/rpa-builder/latest/toolbox-text-recognition-ai-ocr-screen-based

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+
+. Adapt the rest of the Action Steps as you did in the previous Activity.
+
+=== Activity 3: Copy the data from the AMS to 
```
Activity 3: Retrieve and export data from the AMS

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+. Retrieve account data.
+. Retrieve the policy data of the account.
+. Export the data in one of the following ways:
+.. CSV file
```
(in the screenshot, 3b is the CSV export. But we need to check anyway if we need to include the REST Call.)

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+
+Change the following Action Steps:
+
+. To retrieve the data, re-identify the fields from which to read in all the Read Text from App Element Action Steps and add or delete them as needed.
```
https://docs.mulesoft.com/rpa-builder/latest/toolbox-app-automation-read-text-from-app-element

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+image::rpa-setup-optical-character-recognition.png[How to use optical character recognition in RPA Builder]
+
+[start=2]
+. Ensure all retrieved data is pinned to the Combine Strings Action Steps and adjust if necessary. You can also link more Activity Parameters. 
```
https://docs.mulesoft.com/rpa-builder/latest/toolbox-variable-handling-combine-strings

sebermuley replied:
remove s from Action Steps (it's only one)

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,167 @@
+image::rpa-setup-combine-strings-wizard.png[RPA Builder Combine Strings Wizard]
+
+. Add the string to a CSV file via the `Pin Variable`. You can adjust the properties as needed. The CSV file must be in the format specified.
++
```
More info on the CSV format might be found here:

- https://help.salesforce.com/s/articleView?id=000381876&type=1
- https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/datafiles_field_names.htm
- https://developer.salesforce.com/docs/atlas.en-us.248.0.object_reference.meta/object_reference/sforce_api_objects_list.htm
- https://developer.salesforce.com/docs/atlas.en-us.object_reference.meta/object_reference/sforce_api_objects_account.htm

sebermuley replied:
see our project chat

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
Update rpa-setup-guide.adoc

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,176 @@
+
+=== Activity 3: Retrieve and Export Data from the AMS
+
+The third activity retrieves data and exports the data from the AMS to Salesforce Financial Services Cloud.
```
The third activity retrieves data and exports the data from the AMS **so that it can be imported into** Salesforce Financial Services Cloud.
(This will be done by another element of the Accelerator, I think.)

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,176 @@
+
+image::rpa-setup-read-text-tip.png[RPA Builder tip for getting in and out of fields]
+
+If you search for text in a special format, you can use Optical Character Recognition in combination with regular expressions.
```
https://docs.mulesoft.com/rpa-builder/latest/toolbox-text-recognition-ai-ocr-screen-based

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,176 @@
+image::rpa-setup-optical-character-recognition.png[How to use optical character recognition in RPA Builder]
+
+[start=2]
+. Ensure all retrieved data is pinned to the xref:rpa-builder::toolbox-variable-handling-combine-strings.adoc[Combine Strings Action Step] and adjust if necessary. You can also link more Activity Parameters. 
```
https://docs.mulesoft.com/rpa-builder/latest/common-properties#variables [pinned]

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,176 @@
++
+image::rpa-setup-combine-strings-wizard.png[RPA Builder Combine Strings Wizard]
+
+. Add the string to a CSV file via the `Pin Variable`. You can adjust the properties as needed. The CSV file must be in the format specified.
```
Add the string to a CSV file as Pin Variable.

tasmoore committed:
second set of updates from Simone review

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
Update rpa-setup-guide.adoc

Commit:
Update rpa-setup-guide.adoc

Looped in the loop
(or at least tried to do so very quickly)
(caveat: process not tested, requires a bit of research from user)

tasmoore committed:
Update rpa-setup-guide.adoc

sebermuley commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+You can add activities by clicking on an element on the model panel and choosing an element from the context menu. For example, the following image shows how to branch off to a second AMS if you are using more than one and not finding the data you need in the first AMS:
+
+image::rpa-setup-add-activity.png[Add activity to RPA Builder process model]
+
```
Maybe add:
The linter automatically checks the model for completeness and displays messages to indicate what to change.

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+== Before You Begin
+
+* Ensure the Anypoint Platform account that will be used has xref:rpa-home::automation-userrolesandpermissions-anypointrpapermissions.adoc[the RPA Automations Manager and RPA Bots Manager permissions].
```
```suggestion
* Ensure that the Anypoint Platform account to use has xref:rpa-home::automation-userrolesandpermissions-anypointrpapermissions.adoc[the RPA Automations Manager and RPA Bots Manager permissions].
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+* Ensure the Anypoint Platform account that will be used has xref:rpa-home::automation-userrolesandpermissions-anypointrpapermissions.adoc[the RPA Automations Manager and RPA Bots Manager permissions].
+* Install the latest version of MuleSoft xref:rpa-builder::install-and-configure.adoc[RPA Builder] and xref:rpa-bot::installation.adoc[RPA Bot] from the RPA Manager home page.
+* Ensure you have access to your AMS and its database metadata. Configure the custom fields in the [FINS AMS Salesforce Sync Handler - Implementation Template], ensuring that the custom field names match the column names on the CSV file.
```
```suggestion
* Ensure that you have access to your AMS and its database metadata. Configure the custom fields in the [FINS AMS Salesforce Sync Handler - Implementation Template], ensuring that the custom field names match the column names on the CSV file.
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+* Ensure the Anypoint Platform account that will be used has xref:rpa-home::automation-userrolesandpermissions-anypointrpapermissions.adoc[the RPA Automations Manager and RPA Bots Manager permissions].
+* Install the latest version of MuleSoft xref:rpa-builder::install-and-configure.adoc[RPA Builder] and xref:rpa-bot::installation.adoc[RPA Bot] from the RPA Manager home page.
+* Ensure you have access to your AMS and its database metadata. Configure the custom fields in the [FINS AMS Salesforce Sync Handler - Implementation Template], ensuring that the custom field names match the column names on the CSV file.
+* Ensure you have access to the Salesforce Financial Services Cloud organization to which your AMS data will be loaded.
```
```suggestion
* Ensure that you have access to the Salesforce Financial Services Cloud organization to which your AMS data will be loaded.
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+* Install the latest version of MuleSoft xref:rpa-builder::install-and-configure.adoc[RPA Builder] and xref:rpa-bot::installation.adoc[RPA Bot] from the RPA Manager home page.
+* Ensure you have access to your AMS and its database metadata. Configure the custom fields in the [FINS AMS Salesforce Sync Handler - Implementation Template], ensuring that the custom field names match the column names on the CSV file.
+* Ensure you have access to the Salesforce Financial Services Cloud organization to which your AMS data will be loaded.
+* Download the “AMS_to_FSC.crpa” file to the same device on which RPA Builder is installed.
```
```suggestion
* Download the `AMS_to_FSC.crpa` file to the same device on which RPA Builder is installed.
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+== Adapting the CRPA File to Your AMS
+
+. Double-click each activity on the model pane to open the workflow.
+. Compare the process that is shown with your process and adapt the action steps accordingly. See <<example-of-adapting-the-crpa-file>> for a detailed example.
```
```suggestion
. Compare the displayed process with your process and adapt the action steps accordingly. See <<example-of-adapting-the-crpa-file>> for a detailed example.
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+== Implementing and Deploying the First Project
+
+Now that you have published the other project as an invokable run configuration, you can call it from within another process using the xref:rpa-builder::toolbox-network-rest-call.adoc[*REST Call* Action Step].
```
```suggestion
Now that you have published the second project as an invokable run configuration, you can call it from within another process by using the xref:rpa-builder::toolbox-network-rest-call.adoc[*REST Call* Action Step].
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+== Example of Adapting the CRPA File
+
+[NOTE]
+The example in this guide uses real world data to show you how to adapt a CRPA file to a specific AMS.
```
```suggestion
The example in this guide uses real-world data to show you how to adapt a CRPA file to a specific AMS.
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+[NOTE]
+The example in this guide uses real world data to show you how to adapt a CRPA file to a specific AMS.
+
+All values for the Activity Parameters used in RPA Builder are test values that you must overwrite with your own values when creating the run configuration for the process in RPA Manager.
```
Can you break this into two sentences? It is a bit long.

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+[NOTE]
+The example in this guide uses real world data to show you how to adapt a CRPA file to a specific AMS.
+
+All values for the Activity Parameters used in RPA Builder are test values that you must overwrite with your own values when creating the run configuration for the process in RPA Manager.
```
```suggestion
All values for the Activity Parameters used in RPA Builder are test values. You must overwrite these values with your own values when you create the run configuration for the process in RPA Manager.
```

dhirsch1 replied:
The sentence was long so I attempted to break it up.

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+=== Process Model
+
+The process model contains the following three activities:
```
```suggestion
The process model contains the following activities:
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+. Change the following in the xref:rpa-builder::toolbox-control-run-program.adoc[Run Program Action Step]:
+.. Replace the values in the *File name*, *Parameters*, and *Window Title* fields with the values from your AMS.
+.. Click the *Test* button to run the application so that you can use it to adapt the following Action Steps:
```
```suggestion
.. Click *Test* to run the application so that you can use it to adapt the following Action Steps:
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
++
+image::rpa-setup-run-program-wizard.png[RPA Builder Run Program Wizard]
+
+. Use the *Start from scratch* option to capture and test a new pattern that indicates the login page of the AMS is visible onscreen:
```
```suggestion
. Use the *Start from scratch* option to capture and test a new pattern that indicates that the login page of the AMS is visible onscreen:
```

dhirsch1 replied:
Use the *Start from scratch* option > Select *Start from scratch* (if this is something to select).

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
++
+image::rpa-setup-image-search-wizard.png[RPA Builder Image Search Wizard]
+
+. Re-identify the field to enter the password. It may be necessary to add a similar Action Step previously for entering the user name in the xref:rpa-builder::toolbox-app-automation-keystrokes-to-app-element.adoc[Keystrokes to App Element Action Step].
```
```suggestion
. Reidentify the field to enter the password. You might have to add an Action Step for entering the user name in the xref:rpa-builder::toolbox-app-automation-keystrokes-to-app-element.adoc[Keystrokes to App Element Action Step].
```

dhirsch1 replied:
It is not clear what "previously" is referring to. Do you mean the Run Program Action Step?

sebermuley replied:
Hi Diane,
the screenshot shows just one Action Step for entering the password for logging in to the AMS. If the username must be entered, too, the user/reader needs to put another _Keystrokes to App Element_ Action Step in front of/before the password-entering-step in the workflow.

tasmoore replied:
I'll add "Keystrokes to App Element" before "Action Step" to make this clearer.

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
++
+image::rpa-setup-click-app-element-wizard.png[RPA Builder Click App Element Wizard]
+
+. Wait for the login process to complete by choosing a pattern that will disappear after completion:
```
```suggestion
. Wait for the login process to complete by choosing a pattern that disappears after completion:
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+image::rpa-setup-optical-character-recognition.png[How to use optical character recognition in RPA Builder]
+
+[start=2]
+. Ensure all the data that has been retrieved is xref:rpa-builder::common-properties.adoc#variables[pinned] to the xref:rpa-builder::toolbox-variable-handling-combine-strings.adoc[Combine Strings Action Step] and adjust if necessary. You can also link more Activity Parameters. 
```
```suggestion
. Ensure that all retrieved data is xref:rpa-builder::common-properties.adoc#variables[pinned] to the xref:rpa-builder::toolbox-variable-handling-combine-strings.adoc[Combine Strings Action Step] and adjust if necessary. You can also link more Activity Parameters. 
```

dhirsch1 replied:
```suggestion
. Ensure that all retrieved data is xref:rpa-builder::common-properties.adoc#variables[pinned] to the xref:rpa-builder::toolbox-variable-handling-combine-strings.adoc[Combine Strings Action Step] and adjust if necessary. You can also link more Activity Parameters. 
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+== Workflow Overview
+
+. Create two projects in RPA Manager:
+.. The first project contains the process for getting all relevant datasets and looping over them in your AMS. For each dataset, the second project is called.
```
This list item and the next one should be unordered.

dhirsch1 replied:
```suggestion
* The first project contains the process for getting all relevant datasets and looping over them in your AMS. For each dataset, the second project is called.
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+. Create two projects in RPA Manager:
+.. The first project contains the process for getting all relevant datasets and looping over them in your AMS. For each dataset, the second project is called.
+.. The second project contains the process for getting the data from one AMS dataset and adding it to a CSV file, which is imported at a later step to FSC.
```
```suggestion
* The second project contains the process for getting the data from one AMS dataset and adding it to a CSV file, which is imported at a later step to FSC.
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+After successfully testing the project, transfer the project to the Production phase and publish the project to Anypoint Exchange xref:rpa-manager::processautomation-deploy.adoc#publish-automation-exchange[as an invokable run configuration]. Refer to the following documentation for help with this process:
+** xref:rpa-manager::settings-connect.adoc[RPA Manager Settings]
+** https://help.salesforce.com/s/articleView?id=sf.flow_build_use_flows_with_mulesoft_rpa.htm&type=5[Use Flows with MuleSoft RPA]
```
Why does the reference have "RPA Manager Settings" in it?

sebermuley replied:
Could be replaced by 
Connecting to RPA Processes from Salesforce Products

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+== Transferring the Second Project to the Production Phase
+
+After successfully testing the project, transfer the project to the Production phase and publish the project to Anypoint Exchange xref:rpa-manager::processautomation-deploy.adoc#publish-automation-exchange[as an invokable run configuration]. Refer to the following documentation for help with this process:
```
```suggestion
After successfully testing the project, transfer the project to the Production phase and publish the project to Anypoint Exchange xref:rpa-manager::processautomation-deploy.adoc#publish-automation-exchange[as an invokable run configuration]. For help with this process, refer to...```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+Now that you have published the other project as an invokable run configuration, you can call it from within another process using the xref:rpa-builder::toolbox-network-rest-call.adoc[*REST Call* Action Step].
+
+The process should take a list of lookup codes provided via a file and use the xref:rpa-builder::toolbox-flow-control-file-controlled-loop.adoc[*File Controlled Loop* Action Step] to iterate over all codes and use them within the request body of the REST call.
```
```suggestion
The second process should take a list of lookup codes provided via a file, use the xref:rpa-builder::toolbox-flow-control-file-controlled-loop.adoc[*File Controlled Loop* Action Step] to iterate over all codes, and use the codes within the request body of the REST call.
```

sebermuley replied:
Is it clear that the REST Call is made within each iteration of the loop over all lookup codes?

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+image::rpa-setup-process-model.png[RPA Builder process model]
+
+You can add activities by clicking on an element on the model panel and choosing an element from the context menu. For example, the following image shows how to branch off to a second AMS if you are using more than one and not finding the data you need in the first AMS:
```
```suggestion
Add activities by clicking on an element on the model panel and choosing an element from the context menu. For example, the following image shows how to branch off to a second AMS if you are using more than one AMS and not finding the data you need in the first AMS:
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+Click on an activity to open the editing workflow.
+
+=== Activity 1: Log in to the AMS
```
```suggestion
=== Activity 1: Log In to the AMS
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+. Decrypt the AMS login credentials that go in through an Activity Parameter.
+. Run the AMS.
+. Log in to the AMS using the decrypted credentials.
```
```suggestion
. Log in to the AMS by using the decrypted credentials.
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+. Decrypt the AMS login credentials that go in through an Activity Parameter.
+. Run the AMS.
+. Log in to the AMS using the decrypted credentials.
+. Select the database using another in-going Activity Parameter.
```
```suggestion
. Select the database by using another in-going Activity Parameter.
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+=== Activity 2: Find a Record in the AMS
+
+The second activity finds a record in the AMS using a lookup code.
```
```suggestion
The second activity finds a record in the AMS by using a lookup code.
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+The second activity finds a record in the AMS using a lookup code.
+
+. Enter the lookup code, which was provided as an Activity Parameter, as locate criterion to search for the account data set.
+. Click on the result entry to open the account data, and then wait until it is fully loaded.
```
```suggestion
. Click on the result entry to open the account data, and then wait until it's fully loaded.
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+=== Activity 3: Retrieve and Export Data from the AMS
+
+The third activity retrieves data and exports the data from the AMS so that it can be imported into to Salesforce Financial Services Cloud.
```
```suggestion
The third activity retrieves data and exports the data from the AMS so that it can be imported to Salesforce Financial Services Cloud.
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+Change the following Action Steps:
+
+. To retrieve the data, re-identify the fields from which to read in each of the xref:rpa-builder::toolbox-app-automation-read-text-from-app-element.adoc[Read Text from App Element Action Steps] and add or delete them as needed.
```
```suggestion
. To retrieve the data, reidentify the fields from which to read in each of the xref:rpa-builder::toolbox-app-automation-read-text-from-app-element.adoc[Read Text from App Element Action Steps] and add or delete them as needed.
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+image::rpa-setup-read-text-app-element-wizard.png[RPA Builder Read Text from App Element Wizard]
+
+It is sometimes necessary to click into a field before reading the text and then use a keystroke to get out of the field.
```
```suggestion
If necessary, click into a field before reading the text and then use a keystroke to get out of the field.
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
+
+image::rpa-setup-read-text-tip.png[RPA Builder tip for getting in and out of fields]
+
+If you search for text in a special format, you can use xref:rpa-builder::toolbox-text-recognition-ai-ocr-screen-based.adoc[Optical Character Recognition] in combination with regular expressions.
```
```suggestion
To search for text in a special format, you can use xref:rpa-builder::toolbox-text-recognition-ai-ocr-screen-based.adoc[Optical Character Recognition] in combination with regular expressions.
```

dhirsch1 commented on financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc:
```diff
@@ -0,0 +1,183 @@
++
+image::rpa-setup-combine-strings-wizard.png[RPA Builder Combine Strings Wizard]
+
+. Add the string to a CSV file as `Pin Variable`. You can adjust the properties as needed. The CSV file must be in the specified format.
```
```suggestion
. Add the string to a CSV file as `Pin Variable` and adjust the properties as needed. The CSV file must be in the specified format.
```

dhirsch1 reviewed (approved):
The H2s and below should be imperative and not gerunds, according to our style guide. At some point, you might want to think about splitting this up into a few separate topics.

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update financial-services/1.10/modules/ROOT/pages/rpa-setup-guide.adoc

Co-authored-by: Diane E Hirsch <diane.hirsch@mulesoft.com>

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
Update rpa-setup-guide.adoc

tasmoore committed:
Update rpa-setup-guide.adoc
